compiler+runtime/src/cpp/main.cpp

```diff
@@ -184,24 +197,32 @@ namespace jank
       dynamic_call(__rt_ctx->in_ns_var->deref(), make_box<obj::symbol>(opts.target_module));
     }
 
-    /* By default, RL will do tab completion for files. We disable that here. */
-    rl_bind_key('\t', rl_insert);
+    llvm::LineEditor le("jank-native", ".jank-native-repl-history");
+    le.setPrompt("native> ");
+    native_transient_string input{};
 
-    while(auto const buf = readline("native> "))
+    while(std::optional<native_persistent_string> buf = le.readLine())
```

Comment:
I'd still use `auto` here. If we want the type, we can look at the declaration of `le.readLine()`.

Reply:
`auto const &line(*buf);`

This defines a `line` which holds a reference to to the value in the `buf` optional. Is that how we prevent the copies?

Also, it throws an error apparently. No member function `erase` on the `line`